src/NServiceBus.AmazonSQS/MessagePump.cs

```diff
@@ -308,39 +317,50 @@ async Task ProcessMessage(Message receivedMessage, CancellationToken token)
                 catch (Exception ex)
                 {
                     // Can't deserialize. This is a poison message
-                    Logger.Warn($"Treating message with SQS Message Id {receivedMessage.MessageId} as a poison message due to exception {ex}. Moving to error queue.");
+                    exception = ex;
                     isPoisonMessage = true;
                 }
 
                 if (messageBody == null || transportMessage == null)
```

Comment:
can't this be merged into the next condition as:

```
if (isPoisonMessage || messageBody == null || transportMessage == null)
```
```

Reply:
Good catch. Fixed